File: sites/web/src/eyes/skin.ts
import skinsData from "../gen/skins.json"
import type { Cells, Skins, Visual } from "../types"

const skins = skinsData as Skins

export function visual(app: string, cells: Cells, id: number): Visual {
  return skins[app]?.[cells.skin]?.[id] ?? {}
}

export function paint(v: Visual, cells: Cells): string | undefined {
  if (v.bg === undefined) return undefined
  return typeof v.bg === "string" ? v.bg : cells.pens[v.bg.pen]
}

File: pkgs/rs/apps/mrlypaint/src/skin.rs
use mrlyskin::{Skin, Visual};

/// The number of pens the tiles skin expects.
pub const PENS: usize = 16;

/// Builds the tiles skin with one pen visual per pen.
pub fn skin() -> Skin {
    Skin::new((0..PENS).map(Visual::pen).collect())
}

/// Returns the app's skins paired with their names.
pub fn corpus() -> Vec<(&'static str, Skin)> {
    vec![("tiles", skin())]
}
